src/backend/joanie/core/api/client.py

```diff
@@ -1116,26 +1116,29 @@ def download(self, request, pk=None):  # pylint: disable=unused-argument, invali
                 "Cannot get contract when an order is not yet validated."
             )
 
-        if contract.submitted_for_signature_on:
+        was_fully_signed = (
+            contract.student_signed_on
+            and contract.organization_signed_on
+            and contract.submitted_for_signature_on is None
+        )
```

Comment:
Maybe this can be a method in the contract model and you should not use past tense to name your variable